crates/astria-conductor/src/block_cache.rs

```diff
@@ -0,0 +1,220 @@
+//! A cache of sequencer blocks that are only yielded in sequential order.
+use std::{
+    collections::BTreeMap,
+    future::Future,
+};
+
+use pin_project_lite::pin_project;
+use sequencer_client::{
+    tendermint::block::Height,
+    SequencerBlock,
+};
+
+pub(crate) trait GetSequencerHeight {
+    fn get_height(&self) -> Height;
+}
```

Comment:
is this only so it can be re-used with multiple block types? but it's only being used with one block type here, right?

Reply:
Yes, exactly. The "reconstructed block" type that conductor defines and `SequencerBlock` coming out of astria-core both have heights.